### PR TITLE
EZP-30400: Inject TranslationService into NotificationService

### PR DESCRIPTION
**Pull request by mikadamczyk**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZP-30400
| Bug fix?      |no
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | yes
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

To improve developer experience this PR tries simplified using of notifications by removing translator dependency and moving them to the notificationHandler.

Related pull requests: 
- https://github.com/ezsystems/date-based-publisher/pull/103
- https://github.com/ezsystems/ezplatform-page-builder/pull/337
- https://github.com/ezsystems/ezplatform-user/pull/28
- https://github.com/ezsystems/flex-workflow/pull/125

#### Checklist:
- [ ] Coding standards (`$ composer fix-cs`)
- [ ] Ready for Code Review

### CODE REVIEW

File: src/lib/Notification/TranslatableFlashBagNotificationHandler.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Notification;
+
+use Symfony\Component\Translation\TranslatorInterface;
+
+final class TranslatableFlashBagNotificationHandler implements TranslatableNotificationHandlerInterface
```
Comment (adamwojs) [inline src/lib/Notification/TranslatableFlashBagNotificationHandler.php]: Place where notifications  are stored it's implementation detail of `\EzSystems\EzPlatformAdminUi\Notification\NotificationHandlerInterface`:

```suggestion
final class TranslatableNotificationHandler implements TranslatableNotificationHandlerInterface
```

File: src/lib/Notification/TranslatableNotificationHandlerInterface.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Notification;
+
+interface TranslatableNotificationHandlerInterface
```
Comment (adamwojs) [inline src/lib/Notification/TranslatableNotificationHandlerInterface.php]:
```suggestion
/**
 * Generates user notifications with translatable message.
 */
interface TranslatableNotificationHandlerInterface
```

File: src/lib/Notification/TranslatableFlashBagNotificationHandler.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+{
+    /**
+     * @var \EzSystems\EzPlatformAdminUi\Notification\NotificationHandlerInterface
+     */
+    private $notificationHandler;
+    /**
```
Comment (adamwojs) [inline src/lib/Notification/TranslatableFlashBagNotificationHandler.php]: Nitpick: Missing new line between properties.

File: src/lib/Translation/Extractor/NotificationTranslationExtractor.php [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+use JMS\TranslationBundle\Logger\LoggerAwareInterface;
+use JMS\TranslationBundle\Translation\FileSourceFactory;
+use PhpParser\Comment\Doc;
+use PhpParser\Node;
+use PhpParser\NodeTraverser;
+use PhpParser\NodeVisitor;
+use PhpParser\Node\Scalar\String_;
+use Psr\Log\LoggerInterface;
+
+class NotificationTranslationExtractor implements LoggerAwareInterface, FileVisitorInterface, NodeVisitor
```
Comment (adamwojs) [inline src/lib/Translation/Extractor/NotificationTranslationExtractor.php]:
```suggestion
/**
 * Extracts translations from TranslatableNotificationHandler::{info,success,warning,error} method calls. 
 */
final class NotificationTranslationExtractor implements LoggerAwareInterface, FileVisitorInterface, NodeVisitor
```

File: src/lib/Translation/Extractor/NotificationTranslationExtractor.php [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+            || !in_array(strtolower($methodCallNodeName), array_map('strtolower', array_keys($this->methodsToExtractFrom)))) {
+            $this->previousNode = $node;
+
+            return;
+        }
+
+        $ignore = false;
+        $desc = $meaning = null;
+
+        if (null !== $docComment = $this->getDocCommentForNode($node)) {
```
Comment (adamwojs) [inline src/lib/Translation/Extractor/NotificationTranslationExtractor.php]:
```suggestion
        if (null !== ($docComment = $this->getDocCommentForNode($node))) {
```

File: src/lib/Translation/Extractor/NotificationTranslationExtractor.php [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+        }
+
+        if (!$node->args[0]->value instanceof String_) {
+            if ($ignore) {
+                return;
+            }
+
+            $message = sprintf('Can only extract the translation id from a scalar string, but got "%s". Please refactor your code to make it extractable, or add the doc comment /** @Ignore */ to this code element (in %s on line %d).', get_class($node->args[0]->value), $this->file, $node->args[0]->value->getLine());
+
+            if ($this->logger) {
```
Comment (adamwojs) [inline src/lib/Translation/Extractor/NotificationTranslationExtractor.php]: This check is not needed  `$this->logger` is initialized with instance of `\Psr\Log\NullLogger`

File: src/lib/Translation/Extractor/NotificationTranslationExtractor.php [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+        $index = $this->methodsToExtractFrom[strtolower($methodCallNodeName)];
+        if (isset($node->args[$index])) {
+            if (!$node->args[$index]->value instanceof String_) {
+                if ($ignore) {
+                    return;
+                }
+
+                $message = sprintf('Can only extract the translation domain from a scalar string, but got "%s". Please refactor your code to make it extractable, or add the doc comment /** @Ignore */ to this code element (in %s on line %d).', get_class($node->args[$index]->value), $this->file, $node->args[$index]->value->getLine());
+
+                if ($this->logger) {
```
Comment (adamwojs) [inline src/lib/Translation/Extractor/NotificationTranslationExtractor.php]: This check is not needed  `$this->logger` is initialized with instance of `\Psr\Log\NullLogger`

File: src/lib/Notification/TranslatableNotificationHandlerInterface.php [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Notification;
+
+/**
+ * Generates user notifications with translatable message.
```
Comment (pawbuj) [inline src/lib/Notification/TranslatableNotificationHandlerInterface.php]: I think that the description is not needed, the code is clear.

**Pull request merged**